panther_analysis_tool/main.py

```diff
@@ -1204,32 +1202,23 @@ def enrich_test_data(backend: BackendClient, args: argparse.Namespace) -> Tuple[
             return 1, msg
         return 1, "No analysis content to enrich tests data for in {}".format(args.path)
 
-    # Apply the filters as needed
-    if getattr(args, "filter_inverted", None) is None:
-        args.filter_inverted = {}
-    specs = specs.apply(lambda l: filter_analysis(l, args.filter, args.filter_inverted))
+    specs = specs.apply(lambda l: filter_analysis(l, args.analysis_filters))
 
     # If no specs after filtering, nothing to do
     if specs.empty():
         return (
             1,
-            f"No analysis content in {args.path} matched filters {args.filter} - {args.filter_inverted}",
+            f"No analysis content in {args.path} matched filters {args.filter}",
         )
 
     # We only used classify_analysis to figure out what to filter out, if anything.
     # We need to filter our own list of items now, using what's left in the `specs` variable.
     raw_analysis_items_by_id = {}
     for item in raw_analysis_items:
         if item.analysis_spec is not None:
-            rule_id = item.analysis_spec.get("RuleID", "")
-            if rule_id != "":
-                raw_analysis_items_by_id[rule_id] = item
-                continue
-
-            policy_id = item.analysis_spec.get("PolicyID", "")
-            if policy_id != "":
-                raw_analysis_items_by_id[policy_id] = item
-                continue
+            analysis_id = lookup_analysis_id(item.analysis_spec)
+            if analysis_id != UNKNOWN_ANALYSIS_ID:
+                raw_analysis_items_by_id[analysis_id] = item
```

Comment:
we either need a `continue` here, or a slight refactor, ortherwise the below info line will be logged on each loop.

a suggestion:
```
    analysis_id = lookup_analysis_id(item.analysis_spec)
    if analysis_id == UNKNOWN_ANALYSIS_ID:
        logging.info(
            "Analysis item %s is not a Rule, Scheduled Rule, or Policy - skipping",
            item.spec_filename,
        )
        continue

    raw_analysis_items_by_id[analysis_id] = item
```

Reply:
Updated